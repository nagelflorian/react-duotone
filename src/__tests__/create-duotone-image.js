/* global describe, it, expect */
import createDuotoneImage from '../create-duotone-image';

const imageUrl = 'https://avatars.githubusercontent.com/u/7649376?v=3';
const primaryColor = '#66FA75';
const secondaryColor = '#1904C7';

describe('createDuotoneImage', () => {
  document.body.innerHTML = `<div><img id="duotone-image" src="${imageUrl}" /></div>`;

  const imageEl = document.getElementById('duotone-image');

  it('should throw if no colors are passed', () => {
    expect(() => createDuotoneImage(imageEl)).toThrowError(
      'Invalid arguments, You need to pass a primary and secondary color',
    );
  });

  it('should throw if no colors are passed', () => {
    expect(() => createDuotoneImage(imageEl, primaryColor)).toThrowError(
      'Invalid arguments, You need to pass a primary and secondary color',
    );
  });

  it('should return a processed image', () => {
    expect(
      createDuotoneImage(imageEl, primaryColor, secondaryColor),
    ).toMatchSnapshot();
  });
});
