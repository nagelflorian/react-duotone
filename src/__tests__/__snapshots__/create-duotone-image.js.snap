// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createDuotoneImage should return a processed image 1`] = `null`;
